### PR TITLE
Throw ValueError if shapes don't match in keras.layers.Softmax #50467

### PR DESCRIPTION
**Pull request by szutenberg**
I think that we should raise ValueError in case Add is not successful. The reason is that after using tf.keras.layers.Masking, the timestep will be masked (skipped) in all downstream layers (as long as they support masking). The user may be not aware of that.

### CODE REVIEW

**Comment by gbaned** [comment]
It looks like your PR relates to the Keras component. Please submit it to the [github.com/keras-team/keras](github.com/keras-team/keras) repository instead. Thank you.
